src/common/components/dateTimePicker/DateTimePicker.tsx

```diff
@@ -213,6 +213,7 @@ export default function DateTimePicker({
     if (
       process.browser &&
       isOpen &&
+      e.relatedTarget &&
```

Comment:
Should this have some comment explaining why this is needed? Developer might think that this is redundant because it is not easy to see the difference in behaviour of the dropdown. Especially because we don't have automated tests for this.